packages/tokens/non-fungible/src/extensions/consecutive/storage.rs

```diff
@@ -0,0 +1,341 @@
+/// * `token_id` - The identifier of the token to be approved.
+/// * `live_until_ledger` - The ledger number at which the approval expires.
+///
+/// # Errors
+///
+/// * refer to [`self::owner_of`] errors.
+/// * refer to [`crate::storage::approve_for_owner`] errors.
+///
+/// # Events
+///
+/// * topics - `["approve", owner: Address, token_id: TokenId]`
+/// * data - `[approved: Address, live_until_ledger: u32]`
+///
+/// # Notes
+///
+/// * Authorization for `approver` is required.
+pub fn approve(
+    e: &Env,
+    approver: &Address,
+    approved: &Address,
+    token_id: TokenId,
+    live_until_ledger: u32,
+) {
+    approver.require_auth();
+
+    let owner = owner_of(e, token_id);
+    approve_for_owner(e, &owner, approver, approved, token_id, live_until_ledger);
+}
+
+/// Low-level function for handling transfers, mints and burns of an NFT,
+/// without handling authorization. Updates ownership records, adjusts balances,
+/// and clears existing approvals.
+///
+/// The difference with [`crate::storage::update`] is that the current function:
+/// 1. explicitly adds burned tokens to storage in `StorageKey::BurnedToken`,
+/// 2. sets the next token (if any) to the previous owner.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `from` - The address of the current token owner.
+/// * `to` - The address of the token recipient.
+/// * `token_id` - The identifier of the token to be transferred.
+///
+/// # Errors
+///
+/// * [`NonFungibleTokenError::IncorrectOwner`] - If the `from` address is not
+///   the owner of the token.
+/// * refer to [`owner_of`] errors.
+/// * refer to [`increase_balance`] errors.
+pub fn update(e: &Env, from: Option<&Address>, to: Option<&Address>, token_id: TokenId) {
+    if let Some(from_address) = from {
+        let owner = owner_of(e, token_id);
+
+        // Ensure the `from` address is indeed the owner.
+        if owner != *from_address {
+            panic_with_error!(e, NonFungibleTokenError::IncorrectOwner);
+        }
+
+        decrease_balance(e, from_address, 1);
+
+        // Clear any existing approval
+        let approval_key = StorageKey::Approval(token_id);
+        e.storage().temporary().remove(&approval_key);
+
+        // Set the next token to prev owner
+        set_owner_for(e, from_address, token_id + 1);
+    } else {
+        // nothing to do for the `None` case, since we don't track
+        // `total_supply`
+    }
+
+    if let Some(to_address) = to {
+        increase_balance(e, to_address, 1);
+
+        // Set the new owner
+        e.storage().persistent().set(&StorageKey::Owner(token_id), to_address);
+    } else {
+        // Burning: `to` is None
+        e.storage().persistent().remove(&StorageKey::Owner(token_id));
+
+        e.storage().persistent().set(&StorageKey::BurnedToken(token_id), &true);
+    }
+}
+
+/// Low-level function that sets owner of `token_id` to `to` if the token
+/// exists, without handling authorization.
```

Comment:
maybe also mention `if the token exists, and it doesn't have an owner at the moment` or smt along those lines